indigo_app/views/works.py

```diff
@@ -59,23 +60,28 @@ def get_work_timeline(self):
         work_timeline = self.work.points_in_time()
         other_dates = [
             ('assent_date', self.work.assent_date),
-            ('commencement_date', self.work.commencement_date),
             ('publication_date', self.work.publication_date),
             ('repealed_date', self.work.repealed_date)
         ]
-        # add to existing events (e.g. if publication and commencement dates are the same)
-        for entry in work_timeline:
-            for name, date in other_dates:
-                if entry['date'] == date:
-                    entry[name] = True
-        # add new events (e.g. if assent is before any of the other events)
-        existing_dates = [entry['date'] for entry in work_timeline]
+
+        # add new events (e.g. if assent is before any of the other events),
+        # only one event per date
+        def existing_dates():
+            return [entry['date'] for entry in work_timeline]
```

Comment:
If a function is one line (and in this case a simple list comprehension), it's okay not to create a whole new function. You could easily just put this directly into the if statement.

Reply:
This would all be way simpler if work_timeline was a dictionary mapping dates to descriptions!